lib/ui/profitloss/RecordAmount.dart

```diff
@@ -0,0 +1,366 @@
+          selectedDate: viewModel.isEditable
+              ? selectedDate
+              : viewModel.actions[_Constants.firstListItem].selectedDate,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.pickItem,
+          isEditable: viewModel.isEditable,
+          selectedItem: viewModel.isEditable
+              ? selectedCrop
+              : viewModel.actions[_Constants.secondListItem].selectedItem,
+          listOfCrops: viewModel.actions[_Constants.secondListItem].listOfCrops,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.description,
+          isEditable: viewModel.isEditable,
+          description: viewModel.isEditable
+              ? description
+              : viewModel.actions[_Constants.thirdListItem].description,
+        ),
+        style: RecordAmountListItemStyles.biggerStyle,
+        parent: this,
+      ),
+    );
+
+    if (viewModel.isEditable) {
+      listBuilder.add(_buildFooter(viewModel, style));
+    }
+    return listBuilder;
+  }
+
+  Padding _buildFooter(
+      RecordAmountViewModel viewModel, RecordAmountStyle style) {
+    return Padding(
+      padding: style.buttonEdgePadding,
+      child: Row(
+        children: <Widget>[
+          Expanded(
+            child: !widget._viewModel.isFilled
+                ? RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle, onTap: null),
+                    style:
+                        RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                      backgroundColor: Color(0xFFe9eaf2),
+                    ),
+                  )
+                : RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle,
+                        onTap: () => saveData(RecordData(
+                            amount: amount,
+                            date: selectedDate,
+                            crop: selectedCrop,
+                            description: description ?? ""))),
+                    style: viewModel.type == RecordType.sale
+                        ? RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFF24d900),
+                          )
+                        : RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFFff8d4f),
+                          ),
+                  ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void saveData(RecordData save) {
+    final snackBar = SnackBar(
```

Comment:
I think that you didn't understood me 😅  You should pass a function through the ViewModel with a param (RecordData) and return the data selected on the RecordAmount widget to the parent (like the listener that I explained to you last week)

Reply:
we should ask the viewModel to save the data.... viewModel.save(data) in the mock viewmodl just print

Reply:
Remember no temp code in any widgets, we NEVER need to do that since we have a viewModel we delegate to.

Reply:
I added the function in the view model as you commented